src/textual/widgets/text_input.py

```diff
@@ -115,6 +128,7 @@ def __init__(
         *,
         placeholder: str = "",
         initial: str = "",
+        autocompleter: Callable[[str], str | None] | None = None,
```

Comment:
If we want to support async autocomplter (see my comment below) I guess we should extend this type hint to
`Callable[[str], str | None] | Awaitable[[str], str | None] | None = None` ? :-)
(as it starts getting quite long though it might be worth creating a type alias for it :sweat_smile: )